aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -260,6 +182,27 @@ def _validate_payload(self, data: Dict[str, Any], data_record: DataRecord) -> No
             if data_record.payload_hash != data_hash:
                 raise IdempotencyValidationError("Payload does not match stored record for this event key")
 
+    def _validate_hashed_payload(self, old_data_record: DataRecord, data_record: DataRecord) -> None:
+        """
+        Validate that the hashed data provided matches the payload_hash stored data record
+
+        Parameters
+        ----------
+        old_data_record: DataRecord
+            DataRecord instance fetched from Dynamo
+        data_record: DataRecord
+            DataRecord instance which failed insert into Dynamo
+
+        Raises
+        ----------
+        IdempotencyValidationError
+            Payload doesn't match the stored record for the given idempotency key
+
+        """
+        if self.payload_validation_enabled:
+            if old_data_record.payload_hash != data_record.payload_hash:
+                raise IdempotencyValidationError("Hashed payload does not match stored record for this event key")
```

Comment:
We already have a [function to validate](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/idempotency/persistence/base.py#L241) the payload and raise an exception in case the hashed payload does not match the one that is stored. Do you think we can modify the existing function to accept more types/arguments instead of creating a new one?

Reply:
Sure! I've  updated _validate_payload to accept a DataRecord as the payload and removed _validate_hashed_payload:

```python    
def _validate_payload(self, data_payload: Union[Dict[str, Any], DataRecord], stored_data_record: DataRecord) -> None:
        """
        Validate that the hashed payload matches data provided and stored data record

        Parameters
        ----------
        data_payload: Union[Dict[str, Any], DataRecord]
            Payload
        stored_data_record: DataRecord
            DataRecord fetched from Dynamo or cache

        Raises
        ----------
        IdempotencyValidationError
            Payload doesn't match the stored record for the given idempotency key

        """
        if self.payload_validation_enabled:
            if isinstance(data_payload, DataRecord):
                data_hash = data_payload.payload_hash
            else:
                data_hash = self._get_hashed_payload(data=data_payload)

            if stored_data_record.payload_hash != data_hash:
                raise IdempotencyValidationError("Payload does not match stored record for this event key")
```